### PR TITLE
don't start any more client threads while going down

### PR DESCRIPTION
**Pull request by domsj**
I hope this can prevent the failure seen in http://172.19.49.85/view/arakoon-1.7/job/arakoon-1.7-system-tests-slow-SHAKY/173/testReport/junit/arakoon_system_tests.server.shaky/test_restart_master_long/test_restart_master_long/ .

### CODE REVIEW

File: src/tools/server.ml [hunk 1]
```diff
@@ -188,11 +189,17 @@ let make_server_thread
...
-                   Lwt.catch
-                     (fun () -> t)
-                     (fun exn ->
-                        Logger.info_f_ ~exn "Exception in client thread %s" cid))
+                 if not !going_down
+                 then
+                   begin
+                     Hashtbl.add client_threads cid t;
+                     Lwt.catch
+                       (fun () -> t)
```
Comment (toolslive) [inline src/tools/server.ml]: let t = client_thread () in ..... (fun () -> t) 
is a bit heavy as construction.....

File: src/tools/server.ml [hunk 2]
```diff
@@ -188,11 +189,17 @@ let make_server_thread
...
+                       (fun exn ->
+                        Logger.info_f_ ~exn "Exception in client thread %s" cid)
+                   end
+                 else
+                   Lwt.return ())
```
Comment (toolslive) [inline src/tools/server.ml]: maybe we want a variation of 'deny' here.

**Comment by domsj** [comment]
I've tried to address the concerns with a new commit.
Tests are still running, but not expecting any problems there.

**Comment by domsj** [comment]
Hmm, the commit message isn't really correct though .. should be `don't serve any more clients while going down, deny them instead` or sth along those lines.
Will rewrite history with a better message if we think the change should be like this...

**Comment by toolslive** [comment]
ok.

**Pull request merged**